#1  try—catch block
class trycatch
{
  public static void main(String s[])
  {
    int n[]={1,2,3};
    int a=10;
    try
    {
      a=a/10;
      System.out.println(n[3]);
    }   
        catch(ArrayIndexOutOfBoundsException e)
    { 
     System.out.println("out of range");
    }        
     catch(ArithmeticException e)
     {//System.out.println(e.getMessage());
     System.out.println("div by zero");
    }        
     finally
   {    System.out.println("finally quit");
   } }}
……………………………………………………………………………………………………..

#2   Multiple exception in switch case (Unchecked Exceptions)
  class multrycatch
{
  public static void main(String s[])
  {int j;
    for(int i=0;i<=4;i++)
    {           try
         {
          switch(i)
      {case 0: int a=0;
               j=10/a;//ArithmeticException
               break;               
        case 1: char c[]=new char[3];
                 j=c[10];//ArrayIndexOutOfBoundsException
                 break;                 
        case 2: char ch="java".charAt(7);//StringIndexOutOfBoundsException
                 break;            
         case 3:  Integer num=null;//NullPointerException
                 //num = new Integer(10); 
                      num=num+1;  
                      break;
         case 4: String name="10";//NumberFormatException
                 int n=Integer.parseInt(name);
          }//switch        
      }//try
      catch(RuntimeException e)
      {
        System.out.println("case"+i+" ");
        System.out.println(e.getMessage());
      }
    }//for loop
  }//main
}    
……………………………………………………………………………………………………………………. 
#3  CheckedException
import java.io.*;

// Example of FileNotFoundException
class IOTest{
  public static void main(String[] args) throws IOException
  {
    // Creating an instance of FileReader class
    FileReader fileReader = new FileReader("input.txt");
    System.out.println(fileReader.read());
    fileReader.close();
  }
}
